pkg/driver/driver_azure.go

```diff
@@ -359,48 +363,90 @@ func (d *AzureDriver) createVMNicDisk() (*compute.VirtualMachine, error) {
 	}
 	onARMAPISuccess(prometheusServiceSubnet, "subnet.Get")
 
+	/*
+		NIC creation
+	*/
+
 	// Creating NICParameters for new NIC creation request
 	NICParameters := d.getNICParameters(vmName, &subnet)
+
+	// NIC creation request
 	NICFuture, err := clients.nic.CreateOrUpdate(ctx, resourceGroupName, *NICParameters.Name, NICParameters)
 	if err != nil {
+		// Since machine creation failed, delete any infra resources created
+		deleteErr := clients.deleteVMNicDisk(ctx, resourceGroupName, vmName, nicName, diskName)
+		if deleteErr != nil {
+			glog.Errorf("Error occurred during resource clean up: %s", deleteErr)
+		}
+
 		return nil, onARMAPIErrorFail(prometheusServiceNIC, err, "NIC.CreateOrUpdate failed for %s", *NICParameters.Name)
 	}
+
+	// Wait until NIC is created
 	err = NICFuture.WaitForCompletionRef(ctx, clients.nic.Client)
 	if err != nil {
+		// Since machine creation failed, delete any infra resources created
```

Comment:
Is this comment here appropriate or is it intended for NIC ?

Reply:
Since this was a create machine method, i had mentioned machine creation failure. Since the method deletes any infra not just NIC right?